### ISSUE TITLE
Tuned model can't classify new data 

### ISSUE THREAD

**Issue by Noor59007** [NVIDIA/DIGITS#689]
i trained new model using docs here https://github.com/NVIDIA/DIGITS/tree/master/examples/fine-tuning
model trained successful but when i test the model using classify one image is shows  "   IndexError
list index out of range"
here is the console log 
```
210.107.232.130 - - [2016-04-18 14:14:20] "GET / HTTP/1.1" 200 26928 0.022184
210.107.232.130 - - [2016-04-18 14:14:24] "GET /models/20160418-125500-5170 HTTP/1.1" 200 30480 0.012805
2016-04-18 14:14:39 [20160418-141437-65cc] [INFO ] Infer Model task started.
2016-04-18 14:14:40 [20160418-141437-65cc] [WARNING] Infer Model unrecognized output: Couldn't import dot_parser, loading of dot files will not be possible.
2016-04-18 14:14:40 [20160418-141437-65cc] [INFO ] Infer Model task completed.
2016-04-18 14:14:40 [20160418-141437-65cc] [INFO ] Job complete.
2016-04-18 14:14:40 [20160418-141437-65cc] [INFO ] Job deleted.
210.107.232.130 - - [2016-04-18 14:14:40] "POST /models/images/classification/classify_one?job_id=20160418-125500-5170 HTTP/1.1" 500 435 2.758774
2016-04-18 14:19:58 [20160418-141957-fab9] [INFO ] Infer Model task started.
2016-04-18 14:20:00 [20160418-141957-fab9] [WARNING] Infer Model unrecognized output: /home/noor/digits/digits/utils/image.py:108: VisibleDeprecationWarning: using a non-integer number instead of an integer will result in an error in the future
2016-04-18 14:20:00 [20160418-141957-fab9] [WARNING] Infer Model unrecognized output: out[y,x] = image[math.floor(y/ratio), math.floor(x/ratio)]
2016-04-18 14:20:00 [20160418-141957-fab9] [WARNING] Infer Model unrecognized output: Couldn't import dot_parser, loading of dot files will not be possible.
2016-04-18 14:20:00 [20160418-141957-fab9] [INFO ] Infer Model task completed.
2016-04-18 14:20:00 [20160418-141957-fab9] [INFO ] Job complete.
2016-04-18 14:20:00 [20160418-141957-fab9] [INFO ] Job deleted.
210.107.232.130 - - [2016-04-18 14:20:00] "POST /models/images/classification/classify_one?job_id=20160418-125500-5170 HTTP/1.1" 500 435 3.790766
```

**Comment by gheinrich** [comment]
Hello, did you update your model to match the number of outputs with the number of classes? c.f. bug #678? I have pushed PR #682 which should help for that type of issue - can you try it? Thanks!

**Comment by Noor59007** [comment]
@gheinrich  i'm not specifying  num_outouts in last InnerProduct layer here is the code for sample model 

name: "LeNet"
layer {
  name: "mnist"
  type: "Data"
  top: "data"
  top: "label"
  include {
    phase: TRAIN
  }
  data_param {
    batch_size: 64
  }
}
layer {
  name: "mnist"
  type: "Data"
  top: "data"
  top: "label"
  include {
    phase: TEST
  }
  data_param {
    batch_size: 100
  }
}
layer {
  name: "scale"
  type: "Power"
  bottom: "data"
  top: "scale"
  power_param {
    scale: 0.0125
  }
}
layer {
  name: "conv1"
  type: "Convolution"
  bottom: "scale"
  top: "conv1"
  param {
    lr_mult: 0.0
  }
  param {
    lr_mult: 0.0
  }
  convolution_param {
    num_output: 20
    kernel_size: 5
    stride: 1
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "pool1"
  type: "Pooling"
  bottom: "conv1"
  top: "pool1"
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}
layer {
  name: "conv2"
  type: "Convolution"
  bottom: "pool1"
  top: "conv2"
  param {
    lr_mult: 0.0
  }
  param {
    lr_mult: 0.0
  }
  convolution_param {
    num_output: 50
    kernel_size: 5
    stride: 1
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "pool2"
  type: "Pooling"
  bottom: "conv2"
  top: "pool2"
  pooling_param {
    pool: MAX
    kernel_size: 2
    stride: 2
  }
}
layer {
  name: "ip1"
  type: "InnerProduct"
  bottom: "pool2"
  top: "ip1"
  param {
    lr_mult: 0.0
  }
  param {
    lr_mult: 0.0
  }
  inner_product_param {
    num_output: 500
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "relu1"
  type: "ReLU"
  bottom: "ip1"
  top: "ip1"
}
layer {
  name: "ip2"
  type: "InnerProduct"
  bottom: "ip1"
  top: "ip2"
  param {
    lr_mult: 0.0
  }
  param {
    lr_mult: 0.0
  }
  inner_product_param {
    num_output: 10
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "ip2_odd_even"
  type: "InnerProduct"
  bottom: "ip2"
  top: "ip2_odd_even"
  param {
    lr_mult: 1.0
  }
  param {
    lr_mult: 2.0
  }
  inner_product_param {
    weight_filler {
      type: "xavier"
    }
    bias_filler {
      type: "constant"
    }
  }
}
layer {
  name: "accuracy"
  type: "Accuracy"
  bottom: "ip2_odd_even"
  bottom: "label"
  top: "accuracy"
  include { stage: "val" }
}
layer {
  name: "loss"
  type: "SoftmaxWithLoss"
  bottom: "ip2_odd_even"
  bottom: "label"
  top: "loss"
  exclude { stage: "deploy" }
}
layer {
  name: "softmax"
  type: "Softmax"
  bottom: "ip2"
  top: "softmax"
  include { stage: "deploy" }
}

**Comment by gheinrich** [comment]
I see: the `softmax` layer is pointing to the wrong bottom. Can you try the model from https://github.com/NVIDIA/DIGITS/pull/696? Thanks!

**Comment by Noor59007** [comment]
Yes it works
